src/view/form-edit-view.js

```diff
@@ -13,139 +15,292 @@ const createFormEditTemplate = (event) => {
-    this.element.querySelector('form').addEventListener('submit', this.#formSubmitHandler);
-  };
-
-  #formSubmitHandler = (evt) => {
-    evt.preventDefault();
-    this._callback.formSubmit(this.#event);
+    this.element.querySelector('form')
+      .addEventListener('submit', this.#formSubmitHandler);
   };
 
   setDeleteClickHandler = (callback) => {
     this._callback.deleteClick = callback;
-    this.element.querySelector('.event__reset-btn').addEventListener('click', this.#deleteClickHandler);
+    this.element.querySelector('.event__reset-btn')
+      .addEventListener('click', this.#deleteClickHandler);
+  };
+
+  setEditClickHandler = (callback) => {
+    this._callback.editClick = callback;
+    this.element.querySelector('.event__rollup-btn')
+      .addEventListener('click', this.#editClickHandler);
+  };
+
+  reset = (point) => {
+    this.updateElement(
+      FormEditView.parseDataToState(point),
+    );
+  };
+
+  _restoreHandlers = () => {
+    this.#setInnerHandlers();
+    this.setFormSubmitHandler(this._callback.formSubmit);
+    this.setDeleteClickHandler(this._callback.deleteClick);
+    this.setEditClickHandler(this._callback.editClick);
+  };
+
+  #formSubmitHandler = (evt) => {
+    evt.preventDefault();
+    this._callback.formSubmit(FormEditView.parseStateToData(this._state));
   };
 
   #deleteClickHandler = (evt) => {
     evt.preventDefault();
     this._callback.deleteClick();
   };
 
-  setEditClickHandler = (callback) => {
-    this._callback.editClick = callback;
-    this.element.querySelector('.event__rollup-btn').addEventListener('click', this.#editClickHandler);
+  #editClickHandler = (evt) => {
+    evt.preventDefault();
+    this._callback.editClick();
   };
 
-  #editClickHandler = () => {
-    this._callback.editClick();
+  #priceChangeHandler = (evt) => {
+    evt.preventDefault();
+    this._setState({
+      basePrice: evt.target.value
+    });
+  };
+
+  #getChangeDestination = (destinationName) => {
+    const destination = DESTINATIONS.filter((item) => item.name === destinationName);
+
+    if (!destination.length > 0) {
+      return {
+        description: '',
+        name: destinationName,
+        pictures: []
+      };
+    }
+
+    return destination[0];
+  };
+
+  #destinationChangeHandler = (evt) => {
+    evt.preventDefault();
+    this.updateElement({
+      destination: this.#getChangeDestination(evt.target.value)
+    });
+  };
+
+  #typeChangeHandler = (evt) => {
+    evt.preventDefault();
+    const value = evt.target.value;
+
+    if (value) {
+      this.updateElement({
+        type: value,
+        offers: []
+      });
+    }
   };
+
+  #offersChangeHandler = (evt) => {
+    evt.preventDefault();
+    const currentId = Number(evt.target.id.slice(-1));
+    const stateOffers = this._state.offers;
+
+    const updatedOffers = () => stateOffers.includes(currentId)
+      ? stateOffers.filter((item) => item !== currentId)
+      : this._state.offers.concat(currentId);
```

Comment:
Ты сложил this._state.offers в переменную, лучше ее использовать
```suggestion
      ? stateOffers.filter((item) => item !== currentId)
      : stateOffers.concat(currentId);
```